src/twisted/python/_setup.py

```diff
@@ -223,71 +173,25 @@ def getSetupArgs(
-
-    # This is a workaround for distutils behavior; ext_modules isn't
-    # actually used by our custom builder.  distutils deep-down checks
-    # to see if there are any ext_modules defined before invoking
-    # the build_ext command.  We need to trigger build_ext regardless
-    # because it is the thing that does the conditional checks to see
-    # if it should build any extensions.  The reason we have to delay
-    # the conditional checks until then is that the compiler objects
-    # are not yet set up when this code is executed.
-    arguments["ext_modules"] = extensions
+
     # Use custome class to build the extensions.
     class my_build_ext(build_ext_twisted):
         conditionalExtensions = extensions
 
-    command_classes = {"build_ext": my_build_ext, "build_py": BuildPy3}
-
-    requirements = [
-        "zope.interface >= 4.4.2",
-        "constantly >= 15.1",
-        "incremental >= 16.10.1",
-        "Automat >= 0.8.0",
-        "hyperlink >= 17.1.1",
-        "PyHamcrest >= 1.9.0",
-        "attrs >= 19.2.0",
-    ]
-
-    arguments.update(
-        dict(
-            packages=find_packages("src"),
-            use_incremental=True,
-            setup_requires=["incremental >= 16.10.1"],
-            install_requires=requirements,
-            entry_points={"console_scripts": _CONSOLE_SCRIPTS},
-            cmdclass=command_classes,
-            include_package_data=True,
-            exclude_package_data={
-                "": ["*.c", "*.h", "*.pxi", "*.pyx", "build.bat"],
-            },
-            zip_safe=False,
-            extras_require=_EXTRAS_REQUIRE,
-            package_dir={"": "src"},
-        )
-    )
-
-    return arguments
-
-
-class BuildPy3(build_py):
-    """
-    A version of build_py that doesn't install the modules that aren't yet
-    ported to Python 3.
-    """
-
-    def find_package_modules(
-        self, package: str, package_dir: str
-    ) -> List[Tuple[str, str, str]]:
-        modules = [
-            module
-            for module in build_py.find_package_modules(self, package, package_dir)
-            if ".".join([module[0], module[1]]) not in notPortedModules
-        ]
-        return modules
+    return {
+        "long_description": _longDescriptionFromReadme(readme),
+        # This is a workaround for distutils behavior; ext_modules isn't
+        # actually used by our custom builder.  distutils deep-down checks
+        # to see if there are any ext_modules defined before invoking
+        # the build_ext command.  We need to trigger build_ext regardless
+        # because it is the thing that does the conditional checks to see
+        # if it should build any extensions.  The reason we have to delay
+        # the conditional checks until then is that the compiler objects
+        # are not yet set up when this code is executed.
+        "ext_modules": extensions,
+        "cmdclass": {"build_ext": my_build_ext},
+        "extras_require": _EXTRAS_REQUIRE,
```

Comment:
currently _EXTRAS_REQUIRE is in python to save on duplication eg with `twisted[all_non_platform]`